logic.py

```diff
@@ -144,11 +144,13 @@ def get_base_band(level=None):
             return get_base_band()
     else:
         try:
+            default_level = models.SupportLevel.objects.get(default=True)
```

Comment:
There are no db nor runtime constraints preventing more than one row matching this query. Is it something we need to be cautious about?

Reply:
Well, there is a safeguard that prevents more than one being saved as default. This is called on save:

https://github.com/BirkbeckCTP/consortial_billing/blob/022849751e3bef117632ad884b29b887440949ff/logic.py#L253-L266

But it could be that no levels have been set as default, during initial configuration. So I've changed this bit to lean on another function in utils and added a log to make sure it's clear that there needs to be a default for accurate outputs. What do you think?

Reply:
`save()` is fairly weak for a database constraint. For instance bulk operations, such as  `.update()`, can avoid your constraint. A more fault tolerant implementation of the function would be something like:
```diff
diff --git a/logic.py b/logic.py
index 85821d2..a1effcd 100644
--- a/logic.py
+++ b/logic.py
@@ -257,10 +257,6 @@ def keep_default_unique(obj):
     :obj: an unsaved model instance with a property named 'default'
     """
     if obj.default:
-        try:
-            other = type(obj).objects.get(default=True)
-            if obj != other:
-                other.default = False
-                other.save()
-        except type(obj).DoesNotExist:
-            pass
+        type(obj).objects.filter(default=True).exclude(pk=obj.pk).update(
+            default=False,
+        )
```

Reply:
Thanks @mauromsl, but I don't think we'll be writing anything that uses `update()` to make `default` equal `True`. That wouldn't really make sense, because `update()` is typically used for modifying multiple records, and default should be unique. But are you saying I should still defend against it? And how would you do so?

As to the second comment, about refactoring the function, it's not related to the first, is it? The function will still be called on `save()`, not `update()`. So what is the "fault" that this refactor solves?

Reply:
> Thanks @mauromsl, but I don't think we'll be writing anything that uses `update()` to make `default` equal `True`. That wouldn't really make sense, because `update()` is typically used for modifying multiple records, and default should be unique. But are you saying I should still defend against it? And how would you do so?
> 
> As to the second comment, about refactoring the function, it's not related to the first, is it? The function will still be called on `save()`, not `update()`. So what is the "fault" that this refactor solves?

The suggestion on `keep_default_unique` is to guard against a situation where there are already two rows saved with `default` set to `True`, as you did in this PR.

Reply:
OK, I see your thought process now, thanks. I will make this change and then merge.